src/app/child-dev-project/dashboard/dashboard/dashboard-widget.directive.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  ComponentFactoryResolver,
+  Directive,
+  Input,
+  OnChanges,
+  SimpleChanges,
+  ViewContainerRef,
+} from "@angular/core";
+import { LoggingService } from "app/core/logging/logging.service";
+import { DASHBOARD_COMPONENT_MAP } from "../dashboard-component-map";
+import { DashboardWidgetConfig } from "../dashboard-widget-config.interface";
+import { DashboardWidgetComponent } from "../dashboard-widget.component";
+
+/**
+ * Directive to mark a template into which a DashboardWidgetComponent should be loaded.
+ * Pass the DashboardWidgetConfig into the directive to define the widget.
+ */
+@Directive({
+  selector: "[appDashboardWidget]",
+})
+export class DashboardWidgetDirective implements OnChanges {
```

Comment:
generalize this to not only apply to dashboard widgets and move the string-component mapping to the parent